test/app.test.ts

```diff
@@ -1,4 +1,360 @@
+import Runner from 'graasp-test/src/tasks/taskRunner';
+import { StatusCodes } from 'http-status-codes';
+import build from './app';
+import { FIXTURES_INVITATIONS, FIXTURE_ITEM, MockError } from './fixtures';
+import {
+  mockCreateDeleteInvitationTaskSequence,
+  mockCreateTaskSequence,
+  mockCreateUpdateInvitationTaskSequence,
+  mockGetforItemTaskSequence,
+  mockGetTask,
+} from './mocks';
+
+const runner = new Runner();
+const itemMembershipTaskManager = new ItemMembershipTaskManager();
+const memberTaskManager = {
+  getCreateTaskName: jest.fn(),
+} as unknown as MemberTaskManager;
+const itemTaskManager = new ItemTaskManager();
+const item = FIXTURE_ITEM;
+
+describe('Invitation Plugin', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(runner, 'setTaskPostHookHandler').mockImplementation(jest.fn());
+  });
+
+  describe('POST /invite', () => {
+    it('create invitations successfully', async () => {
+      mockCreateTaskSequence(runner, item, FIXTURES_INVITATIONS);
+
+      const mockSendMail = jest.fn().mockImplementation(async () => {
+        // do nothing
+      });
+      const buildInvitationLink = jest.fn().mockReturnValue('link');
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+        sendInvitationEmail: mockSendMail,
+        buildInvitationLink,
+      });
+      const response = await app.inject({
+        method: 'POST',
+        url: `/${item.id}/invite`,
+        payload: { invitations: FIXTURES_INVITATIONS },
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.OK);
+      expect(await response.json()).toEqual(FIXTURES_INVITATIONS);
+
+      // check email got sent
+      setTimeout(() => {
+        expect(mockSendMail).toHaveBeenCalledTimes(FIXTURES_INVITATIONS.length);
+      }, 2000);
+    });
+
+    it('throw if id is invalid', async () => {
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+      });
+      const response = await app.inject({
+        method: 'POST',
+        url: '/invalid-id/invite',
+        payload: { invitations: FIXTURES_INVITATIONS },
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+    });
+
+    it('Return errors correctly', async () => {
+      const resultWithError = [...FIXTURES_INVITATIONS, new MockError()];
+      mockCreateTaskSequence(runner, item, resultWithError);
+
+      const app = await build({
+        runner,
+        itemTaskManager,
+        memberTaskManager,
+        itemMembershipTaskManager,
+      });
+      const response = await app.inject({
+        method: 'POST',
+        url: `/${item.id}/invite`,
+        payload: { invitations: FIXTURES_INVITATIONS },
+      });
+
+      expect(response.statusCode).toEqual(StatusCodes.OK);
+      expect(await response.json()).toEqual(resultWithError);
```

Comment:
Info: why is the status code `OK` if the response contains errors?

Reply:
The array can contain partial errors (one of the invitation failed for example). But now that you point it out, maybe we should go with `207 Multi-Status` but I'm not sure this is standard, what do you think? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#successful_responses

Reply:
I see, in this case I think we should leave it as `200` since according to Wikipedia `207` has a pre-defined shape (RFC 4918) "The message body that follows is by default an [XML](https://en.wikipedia.org/wiki/XML) message and can contain a number of separate response codes, depending on how many sub-requests were made." so this doesn't really fit our use case.